test/matrixnetwork_test.jl

```diff
@@ -25,72 +28,19 @@
     @test is_empty(MatrixNetwork(Int[],Int[],0)) == true
     @test is_empty(MatrixNetwork(Int[1],Int[1])) == false
     @test is_empty(erdos_renyi_undirected(0,0)) == true
-    @test is_empty(erdos_renyi_undirected(1,0)) == false     
-    
+    @test is_empty(erdos_renyi_undirected(1,0)) == false
+
     @testset "matvec" begin
         G = erdos_renyi_directed(5,2)
         y = collect(1:5)
         A = sparse(G)
         @test norm(A*y - G*y,2) <= 10*eps(1.0)
-        @test norm(At_mul_B(G,y) - A'*y,2) <= 10*eps(1.0)        
-    end
-end
-
-@testset "utility" begin
-    @testset "random_edge" begin
-        @test_throws ArgumentError random_edge(empty_graph(0))
-        @test_throws ArgumentError random_edge(empty_graph(5))    
-        A = sparse([1],[2],1.0,5,5)
-        @test random_edge(A)[1:2] == (1,2) 
-        @test random_edge(A)[1:2] == (1,2) 
-        @test random_edge(A)[1:2] == (1,2) 
-        @test random_edge(A)[1:2] == (1,2)             
-        G = lollipop_graph(5,3)
-        A = sparse(G)
-        # there are 16 edges, to 100*16*log(16) should randomly generate all of them
-        M = zeros(8,8)
-        srand(1) # make it determinstic-ish
-        ntrials = 100*16*4
-        for i=1:ntrials # log16 = 4 in base 2
-            M[random_edge(G)[1:2]...] += 1
-        end
-        @test sum(M.*A) == ntrials          # this means we got all 
-        @test all(find(A.*(M .+ A)) .>= 1) # this means we got all entries
-        @test std(M[find(M)]) <= 100
-
-        n = 16
-        A = sparse(1:n-1,2:n,1,n,n)
-        # there are 15 edges, to 100*15*log(15) should randomly generate all of them
-        M = zeros(n,n)
-        srand(1) # make it determinstic-ish
-        ntrials = 100*16*4 # we just use the same one
-        for i=1:ntrials # log16 = 4 in base 2
-            M[random_edge(A)[1:2]...] += 1
-        end
-        @test sum(M.*A) == ntrials          # this means we got all 
-        @test all(find(A.*(M .+ A)) .>= 1) # this means we got all entries
-        @test std(M[find(M)]) <= 100
-    end
-    @testset "undirected_edges" begin
-        @test map(length, undirected_edges(empty_graph(0))) == (0,0) 
-        @test map(length, undirected_edges(empty_graph(1))) == (0,0)     
-        @test map(length, undirected_edges(lollipop_graph(5,1))) == (5,5)
-        @test undirected_edges(lollipop_graph(5,1)) == ([1,2,3,4,5],[2,3,4,5,6])
-        A = sparse([2],[1],1.0,5,5)        
-        @test map(length, undirected_edges(MatrixNetwork(A))) == (0,0)
-    end
-    @testset "directed_edges" begin
-        @test map(length, directed_edges(empty_graph(0))) == (0,0) 
-        @test map(length, directed_edges(empty_graph(1))) == (0,0)     
-        @test map(length, directed_edges(lollipop_graph(5,1))) == (10,10)
-        A = sparse([1],[2],1.0,5,5)
-        @test map(length, directed_edges(MatrixNetwork(A))) == (1,1)
-        @test directed_edges(MatrixNetwork(A)) == ([1],[2])
+        @test norm(G'*y - A'*y,2) <= 10*eps(1.0)
```

Comment:
Where did all the stuff from utility go??

Reply:
Ah, I see now.

Reply:
Resolved